.github/workflows/check-external-contributor.yml

```
@@ -0,0 +1,16 @@
+name: External Fork check
+on:
+  pull_request: {}
+  
+jobs:
+  check-external:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check PR and label
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          if [ '${{ github.event.pull_request.head.repo.fork}}'=='false' ]; then 
+              gh pr edit ${{github.pull_request.head.repo.pulls_url}} --add-label "external"; 
```

Use the pull request number available from the `pull_request` payload [ref](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request)

```suggestion
              gh pr edit ${{ github.event.pull_request.number }} --add-label "external"; 
```